pkg/operator/ceph/controller/controller_utils.go

```diff
@@ -138,22 +139,30 @@ func EnforceHostNetwork() bool {
 }
 
 func SetRevisionHistoryLimit(data map[string]string) {
-	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultValue)
-	if strval != "" {
+	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultStr)
+	var numval int32
```

Comment:
The problem is that `numval` is declared and a value is assigned, but its value is never read.  I would recommend getting rid of the `numval` var and just use the `limit` var when needed. For example, how about this (logging is removed for simplicity).
```go
	strval := k8sutil.GetValue(data, revisionHistoryLimitSettingName, revisionHistoryLimitDefaultStr)
	var limit int32
	if strval == "" {
		limit = revisionHistoryLimitDefault
	} else {
		val, err := strconv.ParseInt(strval, 10, 32)
		if err != nil {
			val = int64(revisionHistoryLimitDefault)
		}
		limit = int32(val)
	}
	revisionHistoryLimit = &limit
```
The last line in my code block is also important to have outside the `else` block, where you had it mistakenly inside.

Reply:
Great, this helped to at least overcome the `go vet`/build problems, but `make test`still fails with  nil pointer deref for me locally 😦 .